### PR TITLE
added /more page for mobile

### LINKED ISSUE #4196: Create app.cal.com/more for mobile menu
We need to reduce the number of items in our menu on mobile. To tackle this we have an overflow menu which will take Routing Forms, Workflows & Settings. 

[Prototype ](https://www.figma.com/proto/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?page-id=15217%3A117835&node-id=15217%3A119635&viewport=1547%2C448%2C0.83&scaling=min-zoom&starting-point-node-id=15217%3A119635&show-proto-sidebar=1) | [View in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=15217%3A118384)

https://user-images.githubusercontent.com/4536123/188568981-7d1fdcd5-fc41-46b4-b526-c7faa26a0ed7.mp4

Image

Image

### PR DESCRIPTION
**Pull request by PeerRich**
https://user-images.githubusercontent.com/8019099/188733486-6b415a39-e5fb-4ca0-a3ad-59bb75fda732.mp4

### CODE REVIEW

File: packages/ui/v2/core/Shell.tsx [hunk 1]
```diff
@@ -399,6 +401,7 @@ const navigation: NavigationItemType[] = [
     name: "Routing Forms",
     href: "/apps/routing_forms/forms",
     icon: Icon.FiFileText,
+    more: true,
```
Comment (PeerRich) [inline packages/ui/v2/core/Shell.tsx:404]: give this "true" and it'll appear in the /more page and not in the bottom nav

File: packages/ui/v2/core/Shell.tsx [hunk 2]
```diff
@@ -407,7 +410,7 @@ const navigation: NavigationItemType[] = [
     name: "workflows",
     href: "/workflows",
     icon: Icon.FiZap,
-    pro: true,
```
Comment (PeerRich) [inline packages/ui/v2/core/Shell.tsx:410]: workflows won't be pro only in 2.0 @CarinaWolli

File: packages/ui/v2/core/Shell.tsx [hunk 3]
```diff
@@ -519,14 +523,20 @@ const MobileNavigation = () => {
...
-          ))}
+          .map((item) => !item.more && <MobileNavigationItem key={item.name} item={item} />)}
+
+        <MobileNavigationItem
+          item={{
+            name: "more",
+            href: "/more",
+            icon: Icon.FiMoreHorizontal,
+          }}
+        />
```
Comment (PeerRich) [inline packages/ui/v2/core/Shell.tsx:539]: this was my best guess at adding a "more" icon without adding it to the desktop navigation. not sure if this is the best approach but it works

File: packages/ui/v2/core/Shell.tsx [hunk 4]
```diff
@@ -551,14 +560,12 @@ const MobileNavigationItem: React.FC<{
     <Link key={item.name} href={item.href}>
       <a
         className={classNames(
-          itemIdx === 0 ? "rounded-l-lg" : "",
-          itemIdx === navigation.length - 1 ? "rounded-r-lg" : "",
-          "group relative min-w-0 flex-1 overflow-hidden bg-white py-2 px-2 text-center text-xs font-medium text-neutral-400 hover:bg-gray-50 hover:text-gray-700 focus:z-10 sm:text-sm [&[aria-current='page']]:text-gray-900"
```
Comment (PeerRich) [inline packages/ui/v2/core/Shell.tsx:556]: we dont need this in the new bottom nav

File: packages/ui/v2/core/Shell.tsx (hunk 3 above)
Comment (zomars) [reply packages/ui/v2/core/Shell.tsx:539]: Will make some adjustments

Comment (PeerRich) [reply packages/ui/v2/core/Shell.tsx:539]: in this PR or followup?

Comment (zomars) [reply packages/ui/v2/core/Shell.tsx:539]: This one. You should be sleeping mate

Comment (zomars) [reply packages/ui/v2/core/Shell.tsx:539]: Made some adjustments, routing forms shouldn't display if not installed 
image

**Review by zomars — APPROVED** [review]
Ship it

File: packages/ui/v2/core/Shell.tsx [hunk 5]
```diff
@@ -428,17 +413,54 @@ const navigation: NavigationItemType[] = [
...
     icon: Icon.FiSettings,
   },
 ];
 
+const MORE_NAME = "more";
+const moreSeparatorIndex = navigation.findIndex((item) => item.name === MORE_NAME);
+const desktopNavigationItems: NavigationItemType[] = [];
+const mobileNavigationBottomItems: NavigationItemType[] = [];
+const mobileNavigationMoreItems: NavigationItemType[] = [];
+navigation.forEach((item, index) => {
```
Comment (alannnc) [inline packages/ui/v2/core/Shell.tsx:446]: one for each to catch'em all @zomars

**Review by alannnc — APPROVED** [review]
Works good!

File: packages/ui/v2/core/Shell.tsx (hunk 5 above)
Comment (zomars) [reply packages/ui/v2/core/Shell.tsx]: @alannnc I take your forEach and up it with reduce

Simpsons Monkey Fight GIF

**Pull request merged**